server/api/events/user_message_events.js

```diff
@@ -0,0 +1,81 @@
+var express = require('express');
+var router = express.Router();
+var Moment = require('moment-timezone');
+const ApiValidator = require("./../api_validator");
+const UserMessageEvents = require("../../models/index").UserMessageEvents;
+const Message = require("../../models/index").Message;
+
+const VALIDATOR = new ApiValidator({
+  state: {
+    defaultValue: 0,
+    maxValue: Message.MESSAGE_STATE_VIEWED,
+    minValue: Message.MESSAGE_STATE_UNCHECKED,
+    type: "integer"
+  }
+});
+
+/* GET events related to a speficic user for a specific message. */
+router.get("/user/:userId/message/:messageId", function(req, res) {	
+  res.jsonResultOf(UserMessageEvents.findReadMessageEventsForUser(req.params.userId, req.params.messageId, { deep: true })
+  .then(function(usermesageevents) {
+    if (!usermesageevents) {
+      throw { status: 404 };
+    }
+    // A message may be viewed only by the sender, the receiver, or an admin.
+    if (!req.isAdmin && req.user.id != usermesageevents.fromUserId && req.user.id != usermesageevents.toUserId) {

```

Comment:
Better to let only the admin view the event log.

This statement would not work anyway - usermesageevents is an array; therefore .fromUserId and .toUserId are always undefined.